pkg/operation/seed/seed.go

```diff
@@ -53,6 +53,68 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// NewBuilder returns a new Builder.
+func NewBuilder() *Builder {
+	return &Builder{
+		seedObjectFunc: func() (*gardencorev1beta1.Seed, error) { return nil, fmt.Errorf("seed object is required but not set") },
+	}
+}
+
+// WithSeedObject sets the seedObject attribute at the Builder.
+func (b *Builder) WithSeedObject(seedObject *gardencorev1beta1.Seed) *Builder {
+	b.seedObjectFunc = func() (*gardencorev1beta1.Seed, error) { return seedObject, nil }
+	return b
+}
+
+// WithSeedObjectFromLister sets the seedObject attribute at the Builder after fetching it from the given lister.
+func (b *Builder) WithSeedObjectFromLister(seedLister gardencorelisters.SeedLister, seedName string) *Builder {
+	b.seedObjectFunc = func() (*gardencorev1beta1.Seed, error) { return seedLister.Get(seedName) }
+	return b
+}
+
+// WithSeedSecret sets the seedSecret attribute at the Builder.
+func (b *Builder) WithSeedSecret(seedSecret *corev1.Secret) *Builder {
```

Comment:
This method is unused.

Reply:
See https://github.com/gardener/gardener/pull/2217#discussion_r417070944